subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -19,6 +21,22 @@ LOG_MODULE_REGISTER(net_sock_addr, CONFIG_NET_SOCKETS_LOG_LEVEL);
 
 #define AI_ARR_MAX	2
 
+#if defined(CONFIG_DNS_RESOLVER) || defined(CONFIG_NET_IPV6)
+#define ANY_RESOLVER
+
+/* Initialize static fields of addrinfo structure. A macro to let it work
+ * with any sockaddr_* type.
+ */
+#define INIT_ADDRINFO(addrinfo, sockaddr) { \
+		(addrinfo)->ai_addr = &(addrinfo)->_ai_addr; \
+		(addrinfo)->ai_addrlen = sizeof(*sockaddr); \
```

Comment:
I know this is not your changes, but just to be in the safe side, the `()` is needed around sockaddr too, so `(addrinfo)->ai_addrlen = sizeof(*(sockaddr));`